pkg/skaffold/gcbreposv2/repo_resolver.go

```diff
@@ -0,0 +1,81 @@
+/*
+Copyright 2024 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package gcbreposv2
+
+import (
+	"context"
+	"fmt"
+
+	cloudbuild "cloud.google.com/go/cloudbuild/apiv2"
+	cloudbuildpb "cloud.google.com/go/cloudbuild/apiv2/cloudbuildpb"
+	"github.com/googleapis/gax-go/v2"
+)
+
+type cloudBuildRepoClient interface {
+	GetRepository(ctx context.Context, req *cloudbuildpb.GetRepositoryRequest, opts ...gax.CallOption) (*cloudbuildpb.Repository, error)
+	FetchReadToken(ctx context.Context, req *cloudbuildpb.FetchReadTokenRequest, opts ...gax.CallOption) (*cloudbuildpb.FetchReadTokenResponse, error)
+	Close() error
+}
+
+var RepositoryManagerClient = repositoryManagerClient
+
+func GetRepoInfo(ctx context.Context, gcpProject, gcpRegion, gcpConnectionName, gcpRepoName string) (repoURI string, readAccessToken string, err error) {
```

Comment:
I'm finding the current design of this function a bit difficult to follow. For a feature requiring multiple PRs, a top-down approach might make it easier to visualize how everything fits together.  Though bottom-up approaches ensure solid unit tests, top-down can be more conducive to getting everyone on the same page – but I understand this preference can be subjective.